x-pack/plugins/security_solution/public/common/components/execution_context/execution_context_propagator.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useExecutionContext } from '@kbn/kibana-react-plugin/public';
+import { useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
+import { HostsTableType } from '../../../hosts/store/model';
+import { UsersTableType } from '../../../users/store/model';
+import { useKibana } from '../../lib/kibana';
+
+const hostsTabs = new Set<string>(Object.values(HostsTableType));
+const usersTabs = new Set<string>(Object.values(UsersTableType));
+
+/**
+ * This component should be used always mounted on the top-level of application.
+ * It utilizes useExecutionContext to make sure all outgoing requests (search,
+ * saved objects) are properly traced back to a page and an entity in ES, APM,
+ * and FullStory.
+ */
+export const ExecutionContextPropagator = () => {
+  const { pathname } = useLocation();
+
+  const normalizedPath = useMemo(() => normalizePath(pathname), [pathname]);
+
+  const { executionContext } = useKibana().services;
+
+  useExecutionContext(executionContext, {
+    type: 'application',
+    page: normalizedPath,
+  });
+
+  return null;
+};
+
+/**
+ * Remove variable parts like UUIDs, IP addresses, user names, etc., from the
+ * given pathname.
+ *
+ * @param pathname url path
+ * @returns url path without variable parts
+ */
+export function normalizePath(pathname: string) {
+  // Replace all uuids in pathname
+  const path = pathname.replace(
+    /\/(?:[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12})/i,
+    '/:uuid'
+  );
+
+  // Sanitize known pages with ids
+  const pathParts = path.split('/');
+  // Normalize routes like /network/ip/1.1.1.1/source to /network/ip/:ip/source
+  if (pathParts[1] === 'network' && pathParts[2] === 'ip' && pathParts[3]) {
+    pathParts[3] = ':ip';
+  }
+  // Normalize routes like /hosts/myHost/authentications to /hosts/:host/authentications
+  if (pathParts[1] === 'hosts' && !hostsTabs.has(pathParts[2])) {
+    pathParts[2] = ':host';
+  }
+  // Normalize routes like /users/myUser/authentications to /users/:user/authentications
+  if (pathParts[1] === 'users' && !usersTabs.has(pathParts[2])) {
+    pathParts[2] = ':user';
```

Comment:
Probably not the best solution as the `normalizePath` method becomes aware of the structure of deeply nested security solution routes. That kind of break incapsulation, but the alternative would be to have many (~70) `ExecutionContextPropagator` components scattered across the entire application under every `<Route>` component.